Sources/Apollo/ApolloClient.swift

```diff
@@ -87,7 +87,9 @@ public class ApolloClient {
   func _fetch<Query: GraphQLQuery>(query: Query, cachePolicy: CachePolicy, context: UnsafeMutableRawPointer? = nil, queue: DispatchQueue, resultHandler: OperationResultHandler<Query>?) -> Cancellable {
     // If we don't have to go through the cache, there is no need to create an operation 
     // and we can return a network task directly
-    if cachePolicy == .fetchIgnoringCacheData {
+    //
+    // But without an operation, we lose the ability to cancel properly.
+    if false, cachePolicy == .fetchIgnoringCacheData {
```

Comment:
? this is hardcoded false

Reply:
It’s the flag I’ve mentioned in my comment above. Removing the `false` reverts the code to the previous *indeterministic* behaviour, in which it doesn't pass the newly added test. With this PR I’m trying to find out how cancelled fetches should deal with the result handler. Once that’s defined, the tests can be updated, and eventually the implementation adjusted. Or we can decide to keep the current behaviour and just improve documentation.